commons/src/main/java/io/aiven/kafka/tieredstorage/commons/UniversalRemoteStorageManager.java

```diff
@@ -214,8 +215,11 @@ private void uploadManifest(final RemoteLogSegmentMetadata remoteLogSegmentMetad
     @Override
     public InputStream fetchLogSegment(final RemoteLogSegmentMetadata remoteLogSegmentMetadata,
                                        final int startPosition) throws RemoteStorageException {
-        return this.fetchLogSegment(remoteLogSegmentMetadata, startPosition,
-            remoteLogSegmentMetadata.segmentSizeInBytes());
+        return this.fetchLogSegment(
+            remoteLogSegmentMetadata,
+            startPosition,
+            remoteLogSegmentMetadata.segmentSizeInBytes() - 1
```

Comment:
Do we need -1 here? AFAICS we are passing this value further where we do exactly the same, or am I wrong?

Reply:
Yes, both will work, but `-1` is technically more correct as the end position is inclusive.